.github/workflows/check-python.yaml

```diff
@@ -109,7 +109,7 @@ jobs:
           PYTHONUTF8: 1
 
       - name: Check diff
-        run: git diff --exit-code examples/ test_cases/
+        run: test -z "$(git status --porcelain examples/ test_cases/)"
 
   compile-all:
```

Comment:
Should also update `check_for_diff` in `test_compile.py`

Reply:
Hol up let me just check this works first, first run should fail.

Reply:
Nah, same check fails on Windows due to line ending changes.
Since this is just to catch accidentally omitted files, I think we just have in the linux compile-all step.